src/main/xsl/preprocess/maprefImpl.xsl

```diff
@@ -30,17 +30,51 @@ See the accompanying LICENSE file for applicable license.
   <xsl:key name="reltable" match="//*[contains(@class, ' map/topicref ')]" use="@dita-ot:orig-format"/>
 
   <xsl:template match="@* | node()">
+    <xsl:param name="relative-path" as="xs:string">#none#</xsl:param>
     <xsl:copy>
-      <xsl:apply-templates select="@* | node()"/>
+      <xsl:apply-templates select="@* | node()">
+        <xsl:with-param name="relative-path" select="$relative-path"/>
+      </xsl:apply-templates>
     </xsl:copy>
   </xsl:template>
 
+  <xsl:template match="@conref">
+    <xsl:param name="relative-path" as="xs:string">#none#</xsl:param>
+    <xsl:attribute name="conref">
+      <xsl:choose>
+        <xsl:when test="$relative-path = ('#none#', '') or starts-with(.,'#')">
+          <xsl:sequence select="."/>
+        </xsl:when>
+        <xsl:otherwise>
+          <xsl:value-of select="dita-ot:normalize-uri(concat($relative-path, .))"/>
+        </xsl:otherwise>
+      </xsl:choose>
+    </xsl:attribute>
+  </xsl:template>
+  
+  <xsl:template match="*[not(contains(@class,' map/topicref '))]/@href">
+    <xsl:param name="relative-path" as="xs:string">#none#</xsl:param>
+    <xsl:attribute name="href">
+      <xsl:choose>
+        <xsl:when test="not(contains(.,'://') or ../@scope = 'external' or $relative-path = ('#none#', ''))">
+          <xsl:value-of select="dita-ot:normalize-uri(concat($relative-path, .))"/>
+        </xsl:when>
+        <xsl:otherwise>
+          <xsl:sequence select="."/>
+        </xsl:otherwise>
+      </xsl:choose>
+    </xsl:attribute>
+  </xsl:template>
+
   <xsl:template match="*[contains(@class, ' map/topicref ')][(@format, @dita-ot:orig-format) = 'ditamap']
                         [empty(@href(: | @dita-ot:orig-href:)) or
                          (:@processing-role = 'resource-only' or:)
                          @scope = ('peer', 'external')]" priority="15">
+    <xsl:param name="relative-path" select="'#none#'" as="xs:string"/>
```

Comment:
Since `$relative-path` is not modified when passing to children, use tunnelled parameter instead.

Reply:
I tried changing to use tunnelled parameters, but it broke other uses. The value changes each time a new submap is encountered, which I'm guessing is the issue. We currently have a mix of tunnelled and not tunnelled uses for `$relative-path` in this module ... seemed like it would be easy to update them all but it broke when I had submaps in 2 different directories.